pkg/segment/writer/segstore.go

```diff
@@ -823,8 +843,37 @@ func (segstore *SegStore) computeStarTree() {
 			segstore.usingSegTree = false
 			return
 		}
+
+		hasTreeSpace := false
+		atreeCounterLock.Lock()
+		if currentAgileTreeCount < MaxConcurrentAgileTrees {
+			// for now the first 5 segstores will get to AgileTree but
+			// we should add some smart logic on how we can rotate this
+			// amongst other indices
+			currentAgileTreeCount++
+			hasTreeSpace = true
```

Comment:
This function is being called from `AppendWipToSegfile`. This function is called when we want to flush the Block. So basically the function `computeStarTree` is being called per block and so the `computeStarTree` will only be executed for `5` blocks. 

Is that what we want? 

I did not understand the comment saying first 5 `segstores`. Are you referring to first 5 segments or first 5 indices?

Reply:
good point. I was missing the case where the segstore (index) has to give back this counter during rotation, so that some other segstore (index) can pick it up 
This function gets called on every wip flush but the creation/initialization of startree happens only on block num 0

Reply:
Ahh, okay I see, the counter is getting incremented only when the `numBlocks` is `zero`. 
But the same segstore (index) can also pick it up again for the next segment.

Also, since `MaxConcurrentAgileTrees` is set to `3` in the latest commit, I think we should update this comment. Instead of stating a specific number like `3` or `5`, refactor the comment to say: `For now, the first MaxConcurrentAgileTrees number of segstores will get to ...`